repo/fsrepo/migrations/retryfetcher.go

```diff
@@ -0,0 +1,43 @@
+package migrations
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+)
+
+type RetryFetcher struct {
+	Fetcher
+	maxRetries int
+}
```

Comment:
Here this might be fine, although generally I've seen people exporting structs protected by initializer functions. I think the main benefit is to do type casting rather than redefining the interface you need, although this seems anti the Go model of trying to declare interfaces at the consumption rather than production points.

Reply:
I could also just scrap the initializer and make this public, WDYT?

Reply:
sure, this is all low risk b/c this has low likelihood of being used anywhere but a handful of places